File: android/app/src/main/java/com/xaymaca/sit/ui/warm/WarmListContainer.kt
package com.xaymaca.sit.ui.warm

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.xaymaca.sit.ui.theme.Warmth
import com.xaymaca.sit.ui.theme.WarmRadius
import com.xaymaca.sit.ui.theme.WarmTheme

/** Cream-paper rounded container with hairline-separated rows. */
@Composable
fun WarmListContainer(
    modifier: Modifier = Modifier,
    warmth: Warmth = Warmth.Subtle,
    content: @Composable () -> Unit,
) {
    val palette = WarmTheme.palette(warmth)
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(WarmRadius.Surface))
            .background(palette.cardBg)
            .border(1.dp, palette.cardBorder, RoundedCornerShape(WarmRadius.Surface)),
    ) {
        content()
    }
}

@Composable
fun WarmRowDivider(
    warmth: Warmth = Warmth.Subtle,
    leadingInset: androidx.compose.ui.unit.Dp = 64.dp,
) {
    val palette = WarmTheme.palette(warmth)
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .padding(start = leadingInset)
            .height(1.dp)
            .background(palette.cardBorder),
    )
}

File: ios/Sources/SIT/Views/Shared/Warm/WarmListContainer.swift
import SwiftUI

/// Cream-paper rounded container with hairline-separated rows. Replaces
/// SwiftUI's default `List` chrome on warm-redesigned screens.
struct WarmListContainer<Content: View>: View {
    var warmth: Warmth = .subtle
    @ViewBuilder var content: Content

    var body: some View {
        let palette = WarmTheme.palette(for: warmth)
        VStack(spacing: 0) {
            content
        }
        .background(palette.cardBg)
        .clipShape(RoundedRectangle(cornerRadius: WarmRadius.surface, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: WarmRadius.surface, style: .continuous)
                .stroke(palette.cardBorder, lineWidth: 1)
        )
    }
}

/// Hairline divider tinted to the warm palette's `cardBorder`. Use between
/// rows inside `WarmListContainer`.
struct WarmRowDivider: View {
    var warmth: Warmth = .subtle

    var body: some View {
        Rectangle()
            .fill(WarmTheme.palette(for: warmth).cardBorder)
            .frame(height: 1)
            .padding(.leading, 64)  // align past leading avatar
    }
}

#Preview {
    WarmListContainer {
        ForEach(0..<3) { i in
            HStack {
                MonogramAvatar(initials: "AB", category: .family)
                VStack(alignment: .leading) {
                    Text("Anna Brown")
                        .font(.system(size: 15, weight: .semibold))
                    Text("Family · last contacted 3d ago")
                        .font(.caption)
                        .foregroundStyle(WarmTheme.subtle.ink2)
                }
                Spacer()
                Image(systemName: "chevron.right")
                    .foregroundStyle(WarmTheme.subtle.ink3)
                    .flipsForRightToLeftLayoutDirection(true)
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 12)
            if i < 2 { WarmRowDivider() }
        }
    }
    .padding()
    .background(WarmTheme.subtle.paper)
}
